### PR TITLE
Adding flatcc dependency and wiring up to bazel/cmake.

### PR DESCRIPTION
**Pull request by benvanik**
This will allow us to generate C flatbuffer reader and verifier files for use in the VM.
Future changes will actually swap out our usage of the C++ version in the VM. The
C++ version will continue to be used for building in the compiler where we don't
care about code size/performance.

Progress on #898.

### CODE REVIEW

File: iree/schemas/BUILD [hunk 1]
```diff
@@ -33,7 +34,16 @@ FLATC_ARGS = [
...
-# TODO(benvanik): also expose as C using flatcc.
+iree_flatbuffer_c_library(
+    name = "bytecode_module_def_c_fbs",
+    srcs = ["bytecode_module_def.fbs"],
+    flatcc_args = [
+        "--reader",
+        "--builder",
+        "--verifier",
+    ],
+)
```
Comment (ScottTodd) [inline iree/schemas/BUILD:45]: Integration on the Google side may be easier if we stage this PR into two changes:
* a first PR that includes all the preceding commits (before "Building bytecode_module_def with flatcc.") and this (and other) usage commented out
* a second PR that flips this to be included

File: iree/vm/BUILD [hunk 2]
```diff
@@ -61,7 +61,9 @@ cc_library(
         "//iree/base:api",
         "//iree/base:flatbuffer_util",
         "//iree/base:target_platform",
+        "//iree/schemas:bytecode_module_def_c_fbs",
         "//iree/schemas:bytecode_module_def_cc_fbs",
+        "@com_github_dvidelabs_flatcc//:runtime",
```
Comment (ScottTodd) [inline iree/vm/BUILD:66]: To help with integration, usage could be deferred to a second PR, or left commented out at first

File: iree/vm/bytecode_module.cc [hunk 3]
```diff
@@ -24,6 +24,11 @@
 #include "iree/vm/ref.h"
 #include "iree/vm/stack.h"
 
+// NOTE: include order matters:
+#include "flatcc/reflection/flatbuffers_common_reader.h"
+#include "iree/schemas/bytecode_module_def_reader.h"
+#include "iree/schemas/bytecode_module_def_verifier.h"
```
Comment (ScottTodd) [inline iree/vm/bytecode_module.cc:30]: Same note about usage and integration here

**Review by ScottTodd — CHANGES_REQUESTED** [review]
Code all LGTM, just have a few reservations about managing this new dependency on the Google side. I started an email thread and dropped a few suggestions for making the process smoother for build cops like @rsuderman .

File: iree/schemas/BUILD (hunk 1 above)
Comment (benvanik) [reply iree/schemas/BUILD]: have reset this to point prior to building bytecode_module_def

**Comment by benvanik** [comment]
Reset to not include actual usage of flatcc. I'm punting on Android for now as I need @antiagainst's assistance to make that work and it's not required.

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**